File: admin/src/api/series.ts
import http from './http'
import type { PageResult, Series, SeriesPostItem } from '@/types/api'

export interface SeriesPayload {
  name: string
  slug?: string
  description?: string
  cover?: string
  visible: boolean
  sort?: number
}

export function getSeriesList(params: {
  page?: number
  pageSize?: number
}): Promise<PageResult<Series>> {
  return http.get('/admin/series', { params })
}

export function createSeries(payload: SeriesPayload): Promise<{ series: Series }> {
  return http.post('/admin/series', payload)
}

export function updateSeries(id: number, payload: SeriesPayload): Promise<{ series: Series }> {
  return http.put(`/admin/series/${id}`, payload)
}

export function deleteSeries(id: number): Promise<null> {
  return http.delete(`/admin/series/${id}`)
}

/** 专题内文章（全部状态，按序号排列） */
export function getSeriesPosts(id: number): Promise<{ list: SeriesPostItem[] }> {
  return http.get(`/admin/series/${id}/posts`)
}

/** 批量调整专题内序号 */
export function reorderSeriesPosts(
  id: number,
  items: { postId: number; sort: number }[],
): Promise<null> {
  return http.put(`/admin/series/${id}/posts`, { items })
}

File: server/model/series.go
package model

// Series 专题（契约：教程型系列文章的内容组织；成员关系存 posts.series_id / series_sort）

import "time"

type Series struct {
	ID          uint   `gorm:"primaryKey" json:"id"`
	Name        string `gorm:"type:varchar(100);uniqueIndex" json:"name"`
	Slug        string `gorm:"type:varchar(100);uniqueIndex" json:"slug"`
	Description string `gorm:"type:varchar(500)" json:"description"`
	Cover       string `gorm:"type:varchar(512)" json:"cover"`
	// Visible 不加 default 标签：GORM 会对带默认值的零值字段做插入省略，false 会被写成 true
	Visible   bool      `json:"visible"`
	Sort      int       `json:"sort"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
}
